doc/manual.rst

```diff
@@ -3564,10 +3564,25 @@ procedures:
 
 
 Procs as expressions can appear both as nested procs and inside top level
-executable code. The  `sugar <sugar.html>`_ module contains the `=>` macro
+executable code. The `sugar <sugar.html>`_ module contains the `=>` macro
 which enables a more succinct syntax for anonymous procedures resembling
 lambdas as they are in languages like JavaScript, C#, etc.
 
+Lambdas
+-------
+
+The `lambdas <lambdas.html>`_ module contains `~>` which enables anonymous
+templates, avoiding some limitations of anonymous procs, in particular, optional
+typing, and offers performance advantage since it enables inlining, unlike
+callbacks which are passed as function pointers.
+
+.. code-block:: nim
+  
+  func isSorted[T, Fun](a: openArray[T], cmp: Fun): bool =
+    for i in 0..<len(a)-1:
+      if not cmp(a[i],a[i+1]): return false
```

Comment:
Since it's ``cmp: Fun`` here, I fail to undersand how this is not a callback here that cannot be inlined.

Reply:
indeed, fixed this description; recent enough C compilers can inline even via function pointers with sufficient optimization level (with -d:release, that may not be the case but should be the case with -d:danger). ~> does however offer more flexibility, eg allowing to omit argument typing (and more generally, alias allows passing uninstantiated generics or overloaded procs, unlike passing via `cmp: Fun`.